src/Microsoft.PowerShell.Commands.Diagnostics/PdhHelper.cs

```diff
@@ -535,22 +535,30 @@ public uint ConnectToDataSource(string dataSourceName)
             return res;
         }
 
-        public uint ConnectToDataSource(StringCollection blgFileNames)
+        public uint ConnectToDataSource(List<string> blgFileNames)
         {
             if (blgFileNames.Count == 1)
             {
                 return ConnectToDataSource(blgFileNames[0]);
             }
 
-            string doubleNullTerminated = string.Empty;
-            foreach (string fileName in blgFileNames)
+            int capacity = blgFileNames.Count + 1;
+            for (int i = 0; i < blgFileNames.Count; i++)
+            {
+                capacity += blgFileNames[i].Length;
+            }
+
+            StringBuilder doubleNullTerminated = new StringBuilder(capacity);
+
+            for (int i = 0; i < blgFileNames.Count; i++)
             {
-                doubleNullTerminated += fileName + '\0';
+                string fileName = blgFileNames[i];
+                doubleNullTerminated.Append(fileName).Append('\0');
             }
 
-            doubleNullTerminated += '\0';
+            doubleNullTerminated.Append('\0');
```

Comment:
Given the way we calculate capacity up front here, it's relatively simple to move to `string.Create()`:

```csharp
string.Create(capacity, blgFileNames, (buf, fileNames) =>
{
    int offset = 0;
    for (int i = 0; i < fileNames.Count; i++)
    {
        string currFileName = fileNames[i];
        int currFileNameLength = currFileName.Length;
        Span<char> currSpan = buf.Slice(offset, currFileNameLength);
        currFileName.AsSpan().CopyTo(currSpan);

        offset += currFileNameLength;
        buf[offset] = '\0';
        offset++;
    }

    buf[offset] = '\0';
});
```

There's probably a bug or two in the above, but I think the general idea is there

Reply:
@xtqqczze let me know if you're interested in trying this out. It's probably not a huge win relatively speaking (compared, for example, to native interop marshalling), but still relevant for an allocation/performance improvement PR

Reply:
@rjmholt We can also refactor the following code doing the same thing: https://github.com/PowerShell/PowerShell/blob/3b83a686a1f1b8811dc0480e2b671a2577333ceb/src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs#L2160-L2188